kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -281,43 +283,58 @@ data IfThenElse term = IfThenElse
     , branch1, branch2 :: !term
     }
 
+data UnifyIfThenElse = UnifyIfThenElse
+    { ifThenElse :: IfThenElse (TermLike RewritingVariableName)
+    , -- The term that was not matched by @matchIfThenElse@
+      term :: TermLike RewritingVariableName
+    }
+
 -- | Match the @KEQUAL.eq@ hooked symbol.
-matchIfThenElse :: TermLike variable -> Maybe (IfThenElse (TermLike variable))
-matchIfThenElse (App_ symbol [condition, branch1, branch2]) = do
-    hook' <- (getHook . symbolHook) symbol
-    Monad.guard (hook' == iteKey)
-    return IfThenElse{symbol, condition, branch1, branch2}
-matchIfThenElse _ = Nothing
+matchIfThenElse ::
```

Comment:
What unification problem does this represent? Could you add documentation in a similar way as you did for the others?